sympy/functions/special/singularity_functions.py

```diff
@@ -0,0 +1,191 @@
+from __future__ import print_function, division
+
+from sympy.core.function import Function, ArgumentIndexError
+from sympy.core import S, sympify, oo, diff
+from sympy.core.logic import fuzzy_not
+from sympy.core.relational import Eq
+from sympy.functions.elementary.complexes import im
+from sympy.functions.elementary.piecewise import Piecewise
+from sympy.functions.special.delta_functions import DiracDelta, Heaviside
+
+###############################################################################
+############################# SINGULARITY FUNCTION ############################
+###############################################################################
+
+
+class SingularityFunction(Function):
+    r"""
+    The Singularity functions are a class of discontinuous functions. It takes a
+    variable , an offset and an exponent as arguments. These functions are
+    represented using Macaulay brackets as :-
+
+    .. math::
+        SingularityFunction(x, a, n) := {\langle x - a \rangle}^ n
```

Comment:
Actually, I have seen docstrings with latex in their main explanation. 

- In [Gamma Functions](https://github.com/sympy/sympy/blob/master/sympy/functions/special/gamma_functions.py#L21)
- In [Beta Functions](https://github.com/sympy/sympy/blob/master/sympy/functions/special/beta_functions.py#L10)
- In [Spherical Harmonics](https://github.com/sympy/sympy/blob/master/sympy/functions/special/spherical_harmonics.py#L16)